### PR TITLE
Honor property::queue::enable_profiling

### PR DESCRIPTION
**Pull request by smaslov-intel**
Signed-off-by: Sergey V Maslov <sergey.v.maslov@intel.com>

### CODE REVIEW

File: SYCL/Plugin/level_zero_queue_profiling.cpp [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+    long long start_time = get_time();
+    sycl::event queue_event = q.submit([&](handler &cgh) {
+      cgh.parallel_for<class empty>(range<2>(10000, 10000),
+                                    [=](item<2> item) { ; });
+    });
+
+    q.wait();
+
+    // Get kernel computation time
+    //    queue_event.wait();
```
Comment (againull) [inline SYCL/Plugin/level_zero_queue_profiling.cpp]: Could you please remove this.

File: SYCL/Plugin/level_zero_queue_profiling.cpp [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+
+// wrapper for gettimeofday
+long long get_time() {
+  struct timeval tv;
+  gettimeofday(&tv, NULL);
+  return (tv.tv_sec * 1000000) + tv.tv_usec;
+}
+
+int foo(queue &q, int n) {
+  for (int i = 0; i < n; i++) {
```
Comment (againull) [inline SYCL/Plugin/level_zero_queue_profiling.cpp:26]: Outer loop and "n" argument look unnecessary, could you please remove it.

File: SYCL/Plugin/level_zero_queue_profiling.cpp [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+
+    long long start_time = get_time();
+    sycl::event queue_event = q.submit([&](handler &cgh) {
+      cgh.parallel_for<class empty>(range<2>(10000, 10000),
+                                    [=](item<2> item) { ; });
```
Comment (againull) [inline SYCL/Plugin/level_zero_queue_profiling.cpp]: NIT: Semicolon is redundant.

File: SYCL/Plugin/level_zero_queue_profiling.cpp [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+// WITH: Kernel time:
+// WITH: Device kernel time:
+// WITH: Device offloading time:
+
+#include <CL/sycl.hpp>
+using namespace cl::sycl;
+#include <sys/time.h>
+
+// wrapper for gettimeofday
+long long get_time() {
```
Comment (againull) [inline SYCL/Plugin/level_zero_queue_profiling.cpp]: get_time() usages look irrelevant to the test. Wouldn't it be enough to test only profiling functionality (Device kernel time)?

File: SYCL/Plugin/level_zero_queue_profiling.cpp (hunk 1 above)
Comment (smaslov-intel) [reply SYCL/Plugin/level_zero_queue_profiling.cpp]: OK

File: SYCL/Plugin/level_zero_queue_profiling.cpp (hunk 2 above)
Comment (smaslov-intel) [reply SYCL/Plugin/level_zero_queue_profiling.cpp:26]: I've changed it to "2" for it to make more sense. In general it may be useful when debugging the test.

File: SYCL/Plugin/level_zero_queue_profiling.cpp (hunk 4 above)
Comment (smaslov-intel) [reply SYCL/Plugin/level_zero_queue_profiling.cpp]: Yes, it would. The reasons that I prefer to leave get_time() in are: 1) it came from a customer so reflects real case better, and 2) it may be helpful for debugging profiling issues, i.e. comparing host clock to that reported by the device.

Comment (smaslov-intel) [reply SYCL/Plugin/level_zero_queue_profiling.cpp]: On the other hand, it makes the test Linux-only. I removed the host timers from the test.

File: SYCL/Plugin/level_zero_queue_profiling.cpp [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+// RUN: env ZE_DEBUG=-1 %GPU_RUN_PLACEHOLDER %t.out 2>&1 | FileCheck --check-prefixes=WITHOUT %s
+// RUN: env ZE_DEBUG=-1 %GPU_RUN_PLACEHOLDER %t.out profile 2>&1 | FileCheck --check-prefixes=WITH %s
+
+// Test case adapted from the SYCL version of Rodinia benchmark hotspot.
+
+// Check the expected output when queue::enable_profiling is not specified
+//
+// WITHOUT: ze_event_pool_desc_t flags set to: 1
+// WITHOUT: terminate called after throwing an instance of
+// 'cl::sycl::runtime_error' WITHOUT:   what():  Native API failed. Native API
```
Comment (againull) [inline SYCL/Plugin/level_zero_queue_profiling.cpp]: Probably a typo or clang-format messed up these lines. If so, then could you please disable clang-format for these lines. I am not sure that test will pass as is.

**Review by againull — DISMISSED** [review]

**Review by vladimirlaz — APPROVED** [review]
